src/sample/courseRegController.java

```diff
@@ -126,6 +126,117 @@ public void displaycourseregistered() {
     }
   }
 
+  public void searchcourse() {
+    // Get the list of courses that can be registered by the student and stored in
+    // an ArrayList named "list"
+    List<String> list = new ArrayList<>();
+    try {
+      File fileObj = new File("data/Student Course List/" + stuID + ".txt");
+      Scanner fileReader = new Scanner(fileObj);
+      while (fileReader.hasNextLine()) {
+        list.add(fileReader.nextLine());
+      }
+      fileReader.close();
+    } catch (FileNotFoundException e) {
+      System.out.println("An error occurred.");
+      e.printStackTrace();
+    }
+
+    // Perform the search(filter) in "list" and output the result in another
+    // ArrayList named "result"
+    List<String> result = list
+        .stream()
+        .filter(x -> x.toLowerCase().contains(searchCourse.getText().toLowerCase()))
+        .collect(Collectors.toList());
+
+    // Convert the elements in "result" from ArrayList to String
+    StringBuilder strbul = new StringBuilder();
+    for (String str : result) {
+      strbul.append(str);
+      strbul.append("\n");
+    }
+
+    String str = strbul.toString();
+
+    // Clear the Course List ListView
+    courseList.getItems().clear();
+
+    // Insert Course List Search Result into txt file
+    try (FileWriter myWriter = new FileWriter("data/Search.txt", true)) {
+      myWriter.write(str);
+    } catch (IOException e) {
+      System.out.println("An error occurred.");
+      e.printStackTrace();
+    }
+    // Display the search result in Listview
+    try {
+      File fileObj = new File("data/Search.txt");
+      Scanner fileReader = new Scanner(fileObj);
+      while (fileReader.hasNextLine()) {
+        courseList.getItems().add(fileReader.nextLine());
+      }
+      fileReader.close();
+      fileObj.delete();
+    } catch (FileNotFoundException e) {
+      System.out.println("An error occurred.");
+      e.printStackTrace();
+    }
+  }
+
+  //Clear the Course Details
+  public void cleardetails(){
+    courseTitle.setText("");
+    time.setText("");
+    desc.setText("");
+  }
+
+  // When a Course in the Course List is Clicked, this function will be executed.
+  public void courselistdetails() {
+    if (courseList.getSelectionModel().getSelectedItem() != null) {
+      addCourseButton.setDisable(false);
```

Comment:
why here disable button?

Reply:
To enable the button again if the user selected a course